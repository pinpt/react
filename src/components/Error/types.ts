import { ReactElement, ReactNode } from 'react';
import { IActionLinkProps } from '../Internal/ActionLink';
import { ILogoProps } from '../Logo';

export interface IErrorProps {
	className?: string;
	logo?: ReactElement<ILogoProps | IActionLinkProps>;
	error?: string;
	title?: string;
	description?: string;
	link?: ReactNode;
}
