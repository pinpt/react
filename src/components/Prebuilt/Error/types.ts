import { ReactElement } from "react";
import { ISite } from "../../../lib";

export interface IPrebuiltErrorProps {
	renderLogo?: (site) => ReactElement;
	site: ISite;
};
