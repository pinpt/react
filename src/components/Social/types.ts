export interface ISocialProps {
	className?: string;
	newTab?: boolean;
	href?: string;
	onClick?: () => void;
}
