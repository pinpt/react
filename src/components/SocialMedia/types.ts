export interface ISocialMediaLinkProps {
	className?: string;
	newTab?: boolean;
	href?: string;
	onClick?: () => void;
}

export interface ISocialMediaShareProps {
	className?: string;
	newTab?: boolean;
	url: string;
	onClick?: () => void;
}
