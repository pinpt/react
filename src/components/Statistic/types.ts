export interface IStatisticProps {
	count: number;
	className?: string;
}
