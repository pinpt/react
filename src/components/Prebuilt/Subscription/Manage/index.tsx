import { ReactElement } from 'react';
import { IHeaderProps } from '../../../Header';
import Header from '../../Header';
import Footer from '../../Footer';
import ManageComponent from '../../../Subscription/Manage';
import { ISite } from '../../../../lib/types/site';
import { ILogoProps } from '../../../Logo';
import { IThemeToggleProps } from '../../../ThemeToggle';
import withWrapper from '../../../Internal/withWrapper';
import { IFooterProps } from '../../../Footer';
import { ISocialMediaBarProps } from '../../../SocialMedia/SocialMediaBar';
import { ICopyrightProps } from '../../../Copyright';

export interface IPrebuiltSubscribeProps {
	className?: string;
	site: ISite;
	renderHeader?: (site: ISite) => ReactElement<IHeaderProps>;
	handleSelectHome?: () => void;
	renderLogo?: (site: ISite) => ReactElement<ILogoProps>;
	renderThemeToggle?: (site: ISite) => ReactElement<IThemeToggleProps>;
	renderFooter?: (site: ISite) => ReactElement<IFooterProps>;
	renderSocial?: (site: ISite) => ReactElement<ISocialMediaBarProps>;
	renderCopyright?: (site: ISite) => ReactElement<ICopyrightProps>;
}

const Subscribe = (props: IPrebuiltSubscribeProps) => {
	const {
		className = '',
		renderHeader,
		site,
		handleSelectHome,
		renderLogo,
		renderThemeToggle,
		renderFooter,
		renderCopyright,
		renderSocial,
	} = props;

	return (
		<div className={`Pinpoint Page SubscriptionSubscribe Prebuilt ${className}`}>
			{withWrapper(
				renderHeader?.(site) ?? (
					<Header
						site={site}
						handleSelectHome={handleSelectHome}
						renderLogo={renderLogo}
						renderSearch={() => <></>}
						renderSubscribe={() => <></>}
						renderThemeToggle={renderThemeToggle}
					/>
				),
				'header'
			)}
			{withWrapper(<ManageComponent />, 'manage')}
			{withWrapper(
				renderFooter?.(site) ?? (
					<Footer
						site={site}
						renderCopyright={renderCopyright}
						renderLogo={renderLogo}
						renderSocial={renderSocial}
						renderSubscribe={() => <></>}
					/>
				),
				'footer'
			)}
		</div>
	);
};

export default Subscribe;
