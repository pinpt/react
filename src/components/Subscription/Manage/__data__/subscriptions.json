{
	"success": true,
	"subscriptions": [
		{
			"subscriptionId": "2611a75dd0b63089",
			"siteId": "XwzQzLbkCGWuC1WHq9Zl",
			"createdAt": "2021-06-24T15:54:50.000Z",
			"subscribed": 1
		},
		{
			"subscriptionId": "84beec9792eff52d",
			"siteId": "PirxVTE94u3YmGNOySRY",
			"createdAt": "2021-06-24T15:15:21.000Z",
			"subscribed": 1
		},
		{
			"subscriptionId": "fd85a8bf7bcbe772",
			"siteId": "8F5TDkf2sJ3jCqSjZnJE",
			"createdAt": "2021-10-21T17:57:35.000Z",
			"subscribed": 1
		}
	],
	"sites": {
		"8F5TDkf2sJ3jCqSjZnJE": {
			"id": "8F5TDkf2sJ3jCqSjZnJE",
			"name": "Demo Site",
			"logoUrl": "fileid:9f827f9dc1c617dc655e791c570e3c74;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
		},
		"PirxVTE94u3YmGNOySRY": {
			"id": "PirxVTE94u3YmGNOySRY",
			"name": "Pinpoint",
			"logoUrl": "fileid:1fcde4196a4c70a8a86f0ce4af53f2a5;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
		},
		"XwzQzLbkCGWuC1WHq9Zl": {
			"id": "XwzQzLbkCGWuC1WHq9Zl",
			"name": "Fish Rules",
			"logoUrl": "fileid:20d3346f1d5905b93b0dd48843716c0e;Ur0D%2CWglf-f%2Cg%25g4e-f5g%2Be-ele-g3e-fRf%2C;1024x1024.png"
		}
	}
}
