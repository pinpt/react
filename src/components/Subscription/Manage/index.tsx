import React, { useMemo } from 'react';
import { faSpinner } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { getFileUrl } from '../../../lib/file_metadata';
import { getRouterAbsolutePath } from '../../../lib/router';
import { ISite } from '../../../lib/types/site';
import { SubscriptionInfo } from '../../../lib/types/subscription';

export interface IManageProps {
	className?: string;
	subscriptions: SubscriptionInfo;
	handleClickUpdate?: (subscriptionId: string) => void;
	handleClickUnsubscribe?: (subscriptionId: string) => void;
	handleClickReSubscribe?: (subscriptionId: string) => void;
	pendingState?: Record<string, boolean>;
	site: ISite;
}
const baseClass = 'Pinpoint SubscriptionManage';
const Manage = (props: IManageProps) => {
	const {
		className = '',
		subscriptions,
		handleClickUpdate,
		handleClickUnsubscribe,
		handleClickReSubscribe,
		pendingState,
		site,
	} = props;

	const filteredSubscriptions = useMemo(() => {
		return subscriptions.subscriptions.filter((sub) => subscriptions.sites[sub.siteId]);
	}, [subscriptions]);

	const fileApi = useMemo(() => {
		if (getRouterAbsolutePath(site, '').includes('edge')) {
			return 'https://file.edge.pinpoint.com';
		}
	}, [site]);

	return (
		<div className={`${baseClass} Wrapper ${className}`}>
			<h2 className={`${baseClass} Title`}>Your Subscriptions</h2>
			<p className={`${baseClass} Description`}>
				Manage your subscriptions below. Missing subscriptions you expect to see? You may have subscribed with a
				different email address.
			</p>
			{filteredSubscriptions.map((sub) => {
				return (
					<div className={`${baseClass} Entry`} key={sub.subscriptionId}>
						<div>
							<div className={`${baseClass} Container`}>
								<img
									className={`${baseClass} SiteLogo`}
									src={getFileUrl(subscriptions.sites[sub.siteId].logoUrl, fileApi)}
								/>
								<span>
									<h2 className={`${baseClass} SiteName`}>{subscriptions.sites[sub.siteId].name}</h2>
									<p className={`${baseClass} SubscriptionDate`}>
										You subscribed {new Date(sub.createdAt).toLocaleString()}
									</p>
								</span>
								<div className={`${baseClass} Buttons`}>
									{sub.subscribed ? (
										<button
											className={`${baseClass} UpdateButton`}
											onClick={() => handleClickUpdate?.(sub.subscriptionId)}
										>
											Update
										</button>
									) : (
										<></>
									)}
									{!!sub.subscribed ? (
										<button
											className={`${baseClass} UnsubscribeButton`}
											onClick={() => handleClickUnsubscribe?.(sub.subscriptionId)}
										>
											{pendingState?.[sub.subscriptionId] ? (
												<FontAwesomeIcon icon={faSpinner} pulse />
											) : (
												'Unsubscribe'
											)}
										</button>
									) : (
										<button
											className={`${baseClass} UpdateButton`}
											onClick={() => handleClickReSubscribe?.(sub.subscriptionId)}
										>
											{pendingState?.[sub.subscriptionId] ? (
												<FontAwesomeIcon icon={faSpinner} pulse />
											) : (
												'Resubscribe'
											)}
										</button>
									)}
								</div>
							</div>
						</div>
					</div>
				);
			})}
		</div>
	);
};

export default Manage;
