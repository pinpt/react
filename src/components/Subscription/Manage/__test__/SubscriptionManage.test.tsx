import renderer from 'react-test-renderer';
import Manage from '..';
import { SubscriptionInfo } from '../../../../lib/types/subscription';
import site from '../../../Prebuilt/__data__/testSite.json';
import subscriptions from '../__data__/subscriptions.json';

test('Test default state', () => {
	const component = renderer.create(
		<Manage
			subscriptions={(subscriptions as unknown) as SubscriptionInfo}
			handleClickReSubscribe={(subId: string) => console.log(`Resubscribe ${subId}`)}
			handleClickUnsubscribe={(subId: string) => console.log(`Unsubscribe ${subId}`)}
			handleClickUpdate={(subId: string) => console.log(`Update ${subId}`)}
			site={site}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test loading state', () => {
	const component = renderer.create(
		<Manage
			subscriptions={(subscriptions as unknown) as SubscriptionInfo}
			handleClickReSubscribe={(subId: string) => alert(`Resubscribe ${subId}`)}
			handleClickUnsubscribe={(subId: string) => alert(`Unsubscribe ${subId}`)}
			handleClickUpdate={(subId: string) => alert(`Update ${subId}`)}
			site={site}
			pendingState={{ '84beec9792eff52d': true }}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
