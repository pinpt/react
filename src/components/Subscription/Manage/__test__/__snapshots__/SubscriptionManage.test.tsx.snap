// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test default state 1`] = `
<div
  className="Pinpoint SubscriptionManage Wrapper "
>
  <h2
    className="Pinpoint SubscriptionManage Title"
  >
    Your Subscriptions
  </h2>
  <p
    className="Pinpoint SubscriptionManage Description"
  >
    Manage your subscriptions below. Missing subscriptions you expect to see? You may have subscribed with a different email address.
  </p>
  <div
    className="Pinpoint SubscriptionManage Entry"
  >
    <div>
      <div
        className="Pinpoint SubscriptionManage Container"
      >
        <img
          className="Pinpoint SubscriptionManage SiteLogo"
          src="/20d3346f1d5905b93b0dd48843716c0e;Ur0D%2CWglf-f%2Cg%25g4e-f5g%2Be-ele-g3e-fRf%2C;1024x1024.png"
        />
        <span>
          <h2
            className="Pinpoint SubscriptionManage SiteName"
          >
            Fish Rules
          </h2>
          <p
            className="Pinpoint SubscriptionManage SubscriptionDate"
          >
            You subscribed 
            6/24/2021, 10:54:50 AM
          </p>
        </span>
        <div
          className="Pinpoint SubscriptionManage Buttons"
        >
          <button
            className="Pinpoint SubscriptionManage UpdateButton"
            onClick={[Function]}
          >
            Update
          </button>
          <button
            className="Pinpoint SubscriptionManage UnsubscribeButton"
            onClick={[Function]}
          >
            Unsubscribe
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Pinpoint SubscriptionManage Entry"
  >
    <div>
      <div
        className="Pinpoint SubscriptionManage Container"
      >
        <img
          className="Pinpoint SubscriptionManage SiteLogo"
          src="/1fcde4196a4c70a8a86f0ce4af53f2a5;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
        />
        <span>
          <h2
            className="Pinpoint SubscriptionManage SiteName"
          >
            Pinpoint
          </h2>
          <p
            className="Pinpoint SubscriptionManage SubscriptionDate"
          >
            You subscribed 
            6/24/2021, 10:15:21 AM
          </p>
        </span>
        <div
          className="Pinpoint SubscriptionManage Buttons"
        >
          <button
            className="Pinpoint SubscriptionManage UpdateButton"
            onClick={[Function]}
          >
            Update
          </button>
          <button
            className="Pinpoint SubscriptionManage UnsubscribeButton"
            onClick={[Function]}
          >
            Unsubscribe
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Pinpoint SubscriptionManage Entry"
  >
    <div>
      <div
        className="Pinpoint SubscriptionManage Container"
      >
        <img
          className="Pinpoint SubscriptionManage SiteLogo"
          src="/9f827f9dc1c617dc655e791c570e3c74;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
        />
        <span>
          <h2
            className="Pinpoint SubscriptionManage SiteName"
          >
            Demo Site
          </h2>
          <p
            className="Pinpoint SubscriptionManage SubscriptionDate"
          >
            You subscribed 
            10/21/2021, 12:57:35 PM
          </p>
        </span>
        <div
          className="Pinpoint SubscriptionManage Buttons"
        >
          <button
            className="Pinpoint SubscriptionManage UpdateButton"
            onClick={[Function]}
          >
            Update
          </button>
          <button
            className="Pinpoint SubscriptionManage UnsubscribeButton"
            onClick={[Function]}
          >
            Unsubscribe
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test loading state 1`] = `
<div
  className="Pinpoint SubscriptionManage Wrapper "
>
  <h2
    className="Pinpoint SubscriptionManage Title"
  >
    Your Subscriptions
  </h2>
  <p
    className="Pinpoint SubscriptionManage Description"
  >
    Manage your subscriptions below. Missing subscriptions you expect to see? You may have subscribed with a different email address.
  </p>
  <div
    className="Pinpoint SubscriptionManage Entry"
  >
    <div>
      <div
        className="Pinpoint SubscriptionManage Container"
      >
        <img
          className="Pinpoint SubscriptionManage SiteLogo"
          src="/20d3346f1d5905b93b0dd48843716c0e;Ur0D%2CWglf-f%2Cg%25g4e-f5g%2Be-ele-g3e-fRf%2C;1024x1024.png"
        />
        <span>
          <h2
            className="Pinpoint SubscriptionManage SiteName"
          >
            Fish Rules
          </h2>
          <p
            className="Pinpoint SubscriptionManage SubscriptionDate"
          >
            You subscribed 
            6/24/2021, 10:54:50 AM
          </p>
        </span>
        <div
          className="Pinpoint SubscriptionManage Buttons"
        >
          <button
            className="Pinpoint SubscriptionManage UpdateButton"
            onClick={[Function]}
          >
            Update
          </button>
          <button
            className="Pinpoint SubscriptionManage UnsubscribeButton"
            onClick={[Function]}
          >
            Unsubscribe
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Pinpoint SubscriptionManage Entry"
  >
    <div>
      <div
        className="Pinpoint SubscriptionManage Container"
      >
        <img
          className="Pinpoint SubscriptionManage SiteLogo"
          src="/1fcde4196a4c70a8a86f0ce4af53f2a5;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
        />
        <span>
          <h2
            className="Pinpoint SubscriptionManage SiteName"
          >
            Pinpoint
          </h2>
          <p
            className="Pinpoint SubscriptionManage SubscriptionDate"
          >
            You subscribed 
            6/24/2021, 10:15:21 AM
          </p>
        </span>
        <div
          className="Pinpoint SubscriptionManage Buttons"
        >
          <button
            className="Pinpoint SubscriptionManage UpdateButton"
            onClick={[Function]}
          >
            Update
          </button>
          <button
            className="Pinpoint SubscriptionManage UnsubscribeButton"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-spinner fa-w-16 fa-pulse "
              data-icon="spinner"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Pinpoint SubscriptionManage Entry"
  >
    <div>
      <div
        className="Pinpoint SubscriptionManage Container"
      >
        <img
          className="Pinpoint SubscriptionManage SiteLogo"
          src="/9f827f9dc1c617dc655e791c570e3c74;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
        />
        <span>
          <h2
            className="Pinpoint SubscriptionManage SiteName"
          >
            Demo Site
          </h2>
          <p
            className="Pinpoint SubscriptionManage SubscriptionDate"
          >
            You subscribed 
            10/21/2021, 12:57:35 PM
          </p>
        </span>
        <div
          className="Pinpoint SubscriptionManage Buttons"
        >
          <button
            className="Pinpoint SubscriptionManage UpdateButton"
            onClick={[Function]}
          >
            Update
          </button>
          <button
            className="Pinpoint SubscriptionManage UnsubscribeButton"
            onClick={[Function]}
          >
            Unsubscribe
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
